### PR TITLE
[202305] update frr patch for bgpd-Ensure-community-data-is-freed-in-some-cases

### PR DESCRIPTION
**Pull request by lipxu**
#### Why I did it
FRR issue
https://github.com/FRRouting/frr/issues/15459
Add patch for FRR PR for 202305
https://github.com/FRRouting/frr/pull/15466

##### Work item tracking
- Microsoft ADO **(number only)**:
- 26876083

#### How I did it
Add patch for FRR PR
https://github.com/FRRouting/frr/pull/15466

#### How to verify it
Run the original case locally

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by yxieca** [comment]
@StormLiangMS the change looks good. but I thought patch 38 can be merged with patch 37 as it has been taken?

**Comment by dgsudharsan** [comment]
> @StormLiangMS the change looks good. but I thought patch 38 can be merged with patch 37 as it has been taken?

Agree with Ying. I would prefer to have a single patch. Can we update?

**Comment by StormLiangMS** [comment]
> > @StormLiangMS the change looks good. but I thought patch 38 can be merged with patch 37 as it has been taken?
> 
> Agree with Ying. I would prefer to have a single patch. Can we update?

File: src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
++++ b/bgpd/bgp_route.c
+@@ -2917,16 +2917,24 @@ void subgroup_process_announce_selected(struct update_subgroup *subgrp,
+ 			 * in FIB, then it is advertised
+ 			 */
+ 			if (advertise) {
+-				if (!bgp_check_withdrawal(bgp, dest))
+-					bgp_adj_out_set_subgroup(
+-						dest, subgrp, &attr, selected);
+-				else
++				if (!bgp_check_withdrawal(bgp, dest)) {
```
Comment (dgsudharsan) [inline src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch:92]: Can you please check the latest update on https://github.com/FRRouting/frr/pull/15466 ? The diff has changed in this file

File: src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+ 						&ri->tx_addpath));
++				bgp_attr_flush(&attr);
+ 			}
+ 		}
+ 	}
+@@ -947,7 +953,7 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
+ 		if (dest) {
+ 			for (pi = bgp_dest_get_bgp_path_info(dest); pi;
+ 			     pi = pi->next) {
+-				if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED))
```
Comment (dgsudharsan) [inline src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch:223]: Can you please check the original PR for this file as well? The diff is little bit different

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch (hunk 1 above)
Comment (lipxu) [reply src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch:92]: > Can you please check the latest update on [FRRouting/frr#15466](https://github.com/FRRouting/frr/pull/15466) ? The diff has changed in this file

HI, @dgsudharsan thanks for reviewing the code. This part of the code should be the same, the difference should be related to the difference between 202305 branch and master. thanks.

File: src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch (hunk 2 above)
Comment (lipxu) [reply src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch:223]: > Can you please check the original PR for this file as well? The diff is little bit different

Yes, this diff was based on our understanding. we will sync with donaldsharp first, and push the latest update if necessary later, thanks.

**Review by dgsudharsan — APPROVED** [review]

**Comment by StormLiangMS** [comment]
PR test pending due to maintainess, bypass the not finished test.

**Pull request merged**